applications/FluidDynamicsApplication/custom_processes/weighted_divergence_calculation_process.h

```diff
@@ -0,0 +1,338 @@
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// Constructor for WeightedDivergenceCalculationProcess Process
+    WeightedDivergenceCalculationProcess(ModelPart& rModelPart
+                    ):
+        Process(),
+        mrModelPart(rModelPart)
+    {
+    }
+
+    /// Destructor.
+    ~WeightedDivergenceCalculationProcess() override {}
+
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+    /// This operator is provided to call the process as a function and simply calls the Execute method.
+    void operator()()
+    {
+        Execute();
+    }
+
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    /// Execution of the process
+    void Execute() override
+    {
+        KRATOS_TRY;
+
+        // Set time coefficient: computations will be performed ONLY AFTER (time_coefficient * END_TIME)
+        const double time_coefficient = 0.2;
+
+        // Extract time information
+        const ProcessInfo& rCurrentProcessInfo = mrModelPart.GetProcessInfo();
+    	const double& time_step_current  = rCurrentProcessInfo[TIME];
+	    const ProcessInfo& rPreviousProcessInfo = rCurrentProcessInfo.GetPreviousTimeStepInfo();
+        const double& time_step_previous = rPreviousProcessInfo[TIME];
+        const double& final_time = rCurrentProcessInfo[END_TIME];
+
+        if (time_step_current >= time_coefficient * final_time) {
+
+            // Check and set number of elements
+            int number_elements;
+            KRATOS_ERROR_IF(mrModelPart.Elements().size() == 0) << "the number of elements in the domain is zero. weighted divergence calculation cannot be applied"<< std::endl;
+            number_elements = mrModelPart.Elements().size();
+
+            // Check and set number of nodes
+            int number_nodes;
+            KRATOS_ERROR_IF(mrModelPart.Nodes().size() == 0) << "the number of nodes in the domain is zero. weighted divergence calculation cannot be applied" << std::endl;
+            number_nodes = mrModelPart.Nodes().size();
+
+            // Current domain size
+            const std::size_t dimension = mrModelPart.GetProcessInfo()[DOMAIN_SIZE];
+
+            // Auxiliar containers
+            Matrix DN_DX, J0;
+            Vector N;
+
+            // Iterate over the elements
+            #pragma omp parallel for firstprivate(DN_DX,  N, J0)
+            for(int i_elem=0; i_elem < number_elements; ++i_elem) {
+                auto it_elem = mrModelPart.ElementsBegin() + i_elem;
+                auto& r_geometry = it_elem->GetGeometry();
+
+                // Current geometry information
+                const std::size_t local_space_dimension = r_geometry.LocalSpaceDimension();
+                // KRATOS_WATCH(local_space_dimension);
+                const std::size_t number_nodes_element = r_geometry.PointsNumber();
+                // KRATOS_WATCH(number_nodes_element);
+
+                // Resize if needed
+                if (DN_DX.size1() != number_nodes_element || DN_DX.size2() != dimension)
```

Comment:
In the FluidDynamicsApplication we normally add the brackets, even for a one line if statements.

Reply:
Ok done